api/src/main/java/org/openmrs/validator/ObsValidator.java

```diff
@@ -77,6 +77,11 @@ public void validate(Object obj, Errors errors) {
 		} else if (obs.getVoided()) {
 			return;
 		}
+		if (obs.getPerson() != null && obs.getEncounter() != null && obs.getEncounter().getPatient() != null) {
```

Comment:
@HerbertYiga , this looks so good to me, May be one more note. Could you possibly concente these !null values for simplicity. i would think like  if (**(obs.getPerson() && obs.getEncounter() && obs.getEncounter().getPatient())!==null)**{
}
Trying to make it a bit logical, how does it look

Reply:
@sherrif10  Thanks much sharif,that changes the whole meaning

Reply:
@HerbertYiga  it doesn't actually changes the meaning! rather the simplified way , however i guess you can simpility that code

Reply:
> @HerbertYiga , this looks so good to me, May be one more note. Could you possibly concente these !null values for simplicity. i would think like if (**(obs.getPerson() && obs.getEncounter() && obs.getEncounter().getPatient())!==null)**{
> }
> Trying to make it a bit logical, how does it look

If `obs.getEncounter()` returns null, calling `obs.getEncounter().getPatient()` will generate NPE. What's in the code is good because it won't let that happen.